core/src/main/scala/org/apache/spark/sql/delta/DeltaSharedExceptions.scala

```diff
@@ -16,13 +16,23 @@
 
 package org.apache.spark.sql.delta
 
+import org.antlr.v4.runtime.ParserRuleContext
+
 import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.parser.{ParseException, ParserUtils}
+import org.apache.spark.sql.catalyst.trees.Origin
 
 class DeltaAnalysisException(
```

Comment:
We add these parameters so we don't lose information in our exceptions thrown in `deltaMerge.scala` that used to use `a.failAnalysis(...)`